packages/mwp-app-render/src/util/seo/metaTags.test.js

```diff
@@ -0,0 +1,71 @@
+import React from 'react';
+import { shallow } from 'enzyme';
+
+import {
+	generateMetaData,
+	generateMetaTags,
+	generateGeoMetaData,
+} from './metaTags';
+
+const MOCK_META = {
+	appPath: '/mock/app/path',
+	baseUrl: 'https://mock-base-url.com',
+	imageUrl: 'https://www.meetup.com/mock-default-image.jpg',
+	localeCode: 'fr-FR',
+	pageTitle: 'mock title',
+	pageDescription: 'mock description',
+	pageKeywords: 'mock,keywords,comma,separated',
+	route: '/mock/preview/path',
+};
+
+describe('generateMetaData', () => {
+	it('matches snap', () => {
+		const metaData = generateMetaData(MOCK_META);
+		expect(metaData).toMatchSnapshot();
+	});
+	it('should exlude meta entries with missing content', () => {
+		const emptyMetaData = generateMetaData([{property: 'mockProp2', content: undefined}, {property: 'mockProp2', content: ''}]);
+		expect(emptyMetaData).toMatchSnapshot();
+	});
+});
+
+describe('generateMetaTags', () => {
+	it('matches snapshot', () => {
+		const metaData = generateMetaData(MOCK_META);
+		const metaTags = generateMetaTags(metaData);
+		expect(metaTags).toMatchSnapshot();
+	});
+});
+
+describe('generateGeoMetaData', () => {
+	const MOCK_GEO = {
+		city: 'mock-city',
+		country: 'mock-country',
+		lat: 'mock-lat',
+		lon: 'mock-lon',
+		state: 'mock-state',
+	};
+	it('matches snap', () => {
+		const geoMetaData = generateGeoMetaData(MOCK_GEO);
+		expect(geoMetaData).toMatchSnapshot();
+	});
+	it('should *not* set geo.position when lat or lon are missing', () => {
+		const mockGeo = { ...MOCK_GEO };
+		delete mockGeo.lat;
+		const geoMetaData = generateGeoMetaData(mockGeo);
+		expect(geoMetaData).toMatchSnapshot();
```

Comment:
since this is a test for the existence of a specific property, a snapshot is probably not the best tool. You can try this to get a very specific failure message:

```js
expect(geoMetaData).not.toContainEqual(expect.objectContaining({ property: "geo.position" }))
```

Jest has a lot of great tools that ensure your tests are as specific as necessary and provide very readable error messages when you use the right matchers: https://facebook.github.io/jest/docs/en/expect.html